assets/src/blocks/Counter/CounterBlock.js

```diff
@@ -1,97 +1,106 @@
 import {frontendRendered} from '../frontendRendered';
+import {CounterFrontend} from './CounterFrontend';
+import {renderToString} from 'react-dom/server';
 
-const BLOCK_NAME = 'planet4-blocks/counter';
+export const BLOCK_NAME = 'planet4-blocks/counter';
 
-export class CounterBlock {
-  constructor() {
-    const {registerBlockType, unregisterBlockStyle, registerBlockStyle} = wp.blocks;
-    const {__} = wp.i18n;
-    const attributes = {
-      title: {
-        type: 'string',
-        default: '',
-      },
-      description: {
-        type: 'string',
-        default: '',
-      },
-      completed: {
-        type: 'integer',
-        default: '',
-      },
-      completed_api: {
-        type: 'string',
-        default: '',
-      },
-      target: {
-        type: 'integer',
-        default: '',
-      },
-      text: {
-        type: 'string',
-        default: '',
-      },
-      style: { // Needed to convert existing blocks
-        type: 'string',
-        default: '',
-      },
-    };
-
-    registerBlockType(BLOCK_NAME, {
-      title: __('Counter', 'planet4-blocks-backend'),
-      icon: 'dashboard',
-      category: 'planet4-blocks',
-      attributes,
-      supports: {
-        html: false, // Disable "Edit as HTMl" block option.
-      },
-      deprecated: [
-        {
-          attributes,
-          save() {
-            return null;
-          },
-        },
-      ],
-      // eslint-disable-next-line no-shadow
-      edit: ({isSelected, attributes, setAttributes}) => {
-        return <CounterEditor
-          attributes={attributes}
-          setAttributes={setAttributes}
-          isSelected={isSelected}
-        />;
-      },
-      save: frontendRendered(BLOCK_NAME),
-    });
+const attributes = {
+  title: {
+    type: 'string',
+    default: '',
+  },
+  description: {
+    type: 'string',
+    default: '',
+  },
+  completed: {
+    type: 'integer',
+    default: '',
+  },
+  completed_api: {
+    type: 'string',
+    default: '',
+  },
+  target: {
+    type: 'integer',
+    default: '',
+  },
+  text: {
+    type: 'string',
+    default: '',
+  },
+  style: { // Needed to convert existing blocks
+    type: 'string',
+    default: '',
+  },
+};
 
-    // Remove the default style since it's the same as "text only"
-    unregisterBlockStyle(BLOCK_NAME, 'default');
+export const registerCounterBlock = () => {
+  const {registerBlockType, unregisterBlockStyle, registerBlockStyle} = wp.blocks;
+  const {RawHTML} = wp.element;
 
-    const styles = [
-      {
-        name: 'plain',
-        label: 'Text Only',
-        isDefault: true,
-      },
+  registerBlockType(BLOCK_NAME, {
+    title: 'Counter',
+    icon: 'dashboard',
+    category: 'planet4-blocks',
+    attributes,
+    supports: {
+      html: false, // Disable "Edit as HTMl" block option.
+    },
+    // eslint-disable-next-line no-shadow
+    edit: CounterEditor,
+    save: ({attributes: saveAttributes}) => {
```

Comment:
```suggestion
    save: props => {
      const markup = renderToString(
        <div
          data-hydrate={BLOCK_NAME}
          data-attributes={JSON.stringify(props.attributes)}
        >
          <CounterFrontend {...props.attributes} />
        </div>
      );
      return <RawHTML>{markup}</RawHTML>;
    },
```

Reply:
I wonder if it's really necessary to destructure props or do something like the above code

Reply:
Sure, both are fine by me, I pushed your suggestion!